MetaMask/metamask-extension
# Add entry point for metalabs dapp into metamask home page

## Explanation

We want to create entry point for the new MetaLabs dapp from the extension. This will give users a quick link to the MetaLabs dapp so they can benefit from the more in-depth portfolio view and other experimental features that the dapp will offer. In addition, we wanted this entry point to be at a location that all users could easily see and access.

For these reasons, I created a subHeader prop in the home component (underneath the tabs) to pass down a react node that displays a link to the metalabs url.

In addition, if the user has not interacted with the page, we will need the feature to request for account permissions from the user with the metalabs url as the origin and redirect to the metalabs page after. There were a couple changes that needed to happen for this to work: firstly, the metalabs url needed to be stored as a query string within the route as the app redirects to the permissions connect pages and secondly, the last page in the permissions connect flow ("permissions-page-container") needed to grab the metalabs url and open a new tab with it.

## More Information

It would be ideal to parameterize the metalabs url because we might want to change the url for a number of reasons in the future. But I wasn't sure what the best mechanism for parameterizing the url was so I went ahead with the most prevalent technique throughout the app: adding the parameter to the `.metamaskrc` file. If there is another way to parameterize the url that you think fits better with what we are doing, please let me know.


## Screenshots/Screencaps
<img width="244" alt="Screen Shot 2022-08-01 at 2 36 10 PM" src="https://user-images.githubusercontent.com/7850658/182250640-20431f14-0298-44ec-a4bb-1475a4487578.png">

Figma designs:
https://www.figma.com/file/gi92L28Nr4PdUaLsTRhmkK/User-Flows---Master?node-id=3%3A1347

## Manual Testing Steps

- Go to home screen on the extension
- Click "portfolio view" link
- Notice that you are redirected to the correct metalabs url

## Pre-Merge Checklist


### + If there are functional changes:
